pkg/reconciler/events/cloudevent/cloudeventsfakeclient.go

```diff
@@ -37,13 +38,17 @@ type FakeClient struct {
 	behaviour *FakeClientBehaviour
 	// Modelled after k8s.io/client-go fake recorder
 	Events chan string
+	// waitGroup is used to block until all events have been sent
+	waitGroup *sync.WaitGroup
 }
 
 // newFakeClient is a FakeClient factory, it returns a client for the target
-func newFakeClient(behaviour *FakeClientBehaviour) cloudevents.Client {
+func newFakeClient(behaviour *FakeClientBehaviour, expectedEventCount int) CEClient {
 	return FakeClient{
 		behaviour: behaviour,
-		Events:    make(chan string, bufferSize),
+		// set buffersize to length of want events to make sure no extra events are sent
```

Comment:
What happens if the controller tried to send more events than expected? Would the test fail?

Reply:
Yes it will fail and this is also the previous behaviour.

Reply:
Oh I see the problem, this will lead to deadlock, I added a buffer size check before sending the event, if extra event wants to be sent it will return error. Plz lmk if that works, thanks!

Reply:
OK - it sounds good - it's important for the test to fail in that case otherwise we would miss error conditions.
Since the test method there is not trivial, I wonder if we should have a test that verifies this case.

Reply:
Sure! I will add a test for this!

Reply:
Added a test for this, ptal! 🙏